### PR TITLE
Add initial AWS account Terraform

### PR DESCRIPTION
**Pull request by BobyMCbobs**
When an account is created, several things must be provisioned on the account (see: #13).
This PR adds reusable Terraform for AWS in an accounts based structure.

# TODO
- [ ] add iam-user module

### CODE REVIEW

**Comment by BobyMCbobs** [comment]
Other account configurations to have via Terraform might include:
- IAM roles
- IAM policies
- IAM user groups

IAM users might be best set up manually. Keen to hear thoughts.

**Comment by BobyMCbobs** [comment]
Added budgeting (556c3ce5ecd541fe7ebf5becc253dee0e3cf8b54) and restructured hierarchically (fbbb927e379a4dc6fd6827f1eb07e4990cfddbc1)

**Comment by BobyMCbobs** [comment]
- Manually created a test account caleb+cncf-credits-terraform-test@ii.coop
- Updated the Terraform to use the account
- `terraform apply`
- IAM group+policy are in-place
- Budget is in-plane
- AuditTrail is running (is filling up the bucket and Terraform detects it's existence) but for some reason not visible in the UI via _OrganizationAccountAccessRole_)

File: infra/aws/terraform/accounts/main.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+variable "alert-emails" {
+  type    = list(string)
+  default = ["jeefy@cncf.io"]
```
Comment (jeefy) [inline infra/aws/terraform/accounts/main.tf:19]: Could we add `projects@cncf.io` as well? I don't want to be a single point of failure. :)

**Review by jeefy — CHANGES_REQUESTED** [review]
Quick change re: alert emails. Looks good so far!

File: infra/aws/terraform/accounts/main.tf (hunk 1 above)
Comment (BobyMCbobs) [reply infra/aws/terraform/accounts/main.tf]: Absolutely, let's fix that! Added in 7833134

**Comment by BobyMCbobs** [comment]
I'd like to merge this one soon to start rolling out infra.

Looking for some more review or approval.

cc @jeefy @hh

**Comment by BobyMCbobs** [comment]
I added a ceiling budget alert as well as a (defaulting to) 75% budget alert.

**Comment by BobyMCbobs** [comment]
Thank you @jeefy!

**Pull request merged**